### PR TITLE
[e2e] Creates basic support/documentation/example to iOS.

### PR DESCRIPTION
**Pull request by AlbertWang0116**
## Description

    * Updates iOS E2EPlugin and add skeleton iOS test case E2EIosTest.
    * Adds instructions to README.md about e2e testing on iOS device.
    * Applies iOS e2e to example.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Comment by AlbertWang0116** [comment]
@collinjackson

File: packages/e2e/README.md [hunk 1]
```diff
@@ -151,4 +168,36 @@ You can pass additional parameters on the command line, such as the
 devices you want to test on. See
 [gcloud firebase test android run](https://cloud.google.com/sdk/gcloud/reference/firebase/test/android/run).
 
-iOS support for Firebase Test Lab is not yet available, but is planned.
+## iOS device testing
+
+You need to change `iOS/Podfile` to avoid test target statically linking to the plugins. One way is to
+link all of the plugins dynamically:
```
[the remark this answers is not in the document]
Comment (AlbertWang0116) [reply packages/e2e/README.md]: [1] https://stackoverflow.com/a/45457045
[2] https://stackoverflow.com/a/46933115

File: packages/e2e/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,8 @@
+## 0.2.2+4
+* Updates E2EPlugin and add skeleton iOS test case E2EIosTest.
```
Comment (collinjackson) [inline packages/e2e/CHANGELOG.md]:
```suggestion
* Updates `E2EPlugin` and add skeleton iOS test case `E2EIosTest`.
```

File: packages/e2e/CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,8 @@
...
+* Adds instructions to README.md about e2e testing on iOS device.
```
Comment (collinjackson) [inline packages/e2e/CHANGELOG.md]:
```suggestion
* Adds instructions to README.md about e2e testing on iOS devices.
```

File: packages/e2e/CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,8 @@
...
+* Applies iOS e2e to example.
```
Comment (collinjackson) [inline packages/e2e/CHANGELOG.md]:
```suggestion
* Adds iOS e2e testing to example.
```

File: packages/e2e/README.md [hunk 5]
```diff
@@ -27,6 +27,23 @@ void main() {
 }
 ```
 
+### Additional steps for iOS
+
+iOS e2e package depends on XCTest framework, which requires your project to
```
Comment (collinjackson) [inline packages/e2e/README.md]:
```suggestion
iOS e2e package depends on the XCTest framework, which requires your project to
```

File: packages/e2e/README.md [hunk 6]
```diff
@@ -151,4 +168,36 @@ You can pass additional parameters on the command line, such as the
...
 
-iOS support for Firebase Test Lab is not yet available, but is planned.
+## iOS device testing
+
+You need to change `iOS/Podfile` to avoid test target statically linking to the plugins. One way is to
+link all of the plugins dynamically:
+
+```
+target 'Runner' do
+  use_framework!
```
Comment (collinjackson) [inline packages/e2e/README.md]:
```suggestion
  use_frameworks!
```

I think there's an "s" here but maybe I'm mistaken.

File: packages/e2e/ios/Classes/E2EIosTest.m [hunk 7]
```diff
@@ -0,0 +1,30 @@
+#import "E2EIosTest.h"
+#import "E2EPlugin.h"
+
+@implementation E2EIosTest
+
+- (void)testStub {
```
Comment (collinjackson) [inline packages/e2e/ios/Classes/E2EIosTest.m]:
```suggestion
- (void)testE2E {
```

I like this name better but don't feel strongly about it.

**Comment by AlbertWang0116** [comment]
I made E2EIosTest fallback not to be a XCTestCase class for now. Otherwise iOS app won't run until developer manually open .xcodeproj and run a test case. Flutter tool should be able to resolve this, then we can switch E2EIosTest to XCTestCase.

File: packages/e2e/CHANGELOG.md [hunk 8]
```diff
@@ -1,3 +1,8 @@
+## 0.2.2+4
```
Comment (collinjackson) [inline packages/e2e/CHANGELOG.md]:
```suggestion
## 0.2.3

```

**Pull request merged**